sys/net/gnrc/conn/gnrc_conn.c

```diff
@@ -38,7 +75,7 @@ int gnrc_conn_recvfrom(conn_t *conn, void *data, size_t max_len, void *addr, siz
                 }
                 LL_SEARCH_SCALAR(pkt, l3hdr, type, conn->l3_type);
                 if (l3hdr == NULL) {
-                    msg_send_to_self(&msg); /* requeue invalid messages */
+                    msg_send_to_self(&msg); /* requeue invalid messages, wouldn't this make the thread just read the message 3 times and return? */
```

Comment:
No, if there are less than 3 messages in the queue, it will indeed be checked multiple times. IMO the 3 is rather arbitrary anyway and we need a better concept here in the future. For now I don't see a better solution.